packages/waas/src/secure-store.ts

```diff
@@ -0,0 +1,59 @@
+import { openDB } from 'idb'
+
+export interface SecureStoreBackend {
+    get(dbName: string, dbStoreName: string, key: string): Promise<string | null>
+    set(dbName: string, dbStoreName: string, key: string, value: string): Promise<boolean>
+    delete(dbName: string, dbStoreName: string, key: string): Promise<boolean>
+}
+
+export const getDefaultSecureStoreBackend = (): SecureStoreBackend | null => {
+    if (isIndexedDbAvailable()) {
+        return new IndexedDbSecureStoreBackend()
+    } else {
+        return null
+    }
+}
+
+export function isIndexedDbAvailable(): boolean {
+    return typeof indexedDB === 'object'
```

Comment:
btw, is it intentional to not use `window.indexedDB` here..? (just curious)

Reply:
Yes, intentional. This is how `idb` references it as well. Using `window.indexedDB` doesn't seem to work (at least when running tests)